include/shell/shell.h

```diff
@@ -137,7 +126,7 @@ struct shell_static_entry {
 			STRINGIFY(UTIL_CAT(shell_root_cmd_, syntax)))))	   \
 	__attribute__((used)) = {					   \
 		.is_dynamic = false,					   \
```

Comment:
This will not work for C++. You cannot use structure initializer that name the fields, ```.is_dynamic = false,``` will not compile. The better way to keep it running in both C and C++ is to set directly like ```__attribute__((used)) = {					   
		 false, {&UTIL_CAT(_shell_, syntax)}} ``` maybe you'll need a cast to the first union field.